code/script.js

```diff
@@ -231,95 +233,138 @@ const setSecret = () => {
+    }  
+  } else if (category === 'hair'){
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} ${category}! Keeping all people that have ${value} ${category}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    } else {
+      alert(
+        `No, the person doesn't have ${value} ${category}! Removing all people that have ${value} ${category}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
+  } else if (category === 'eyes') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} ${category}! Keeping all people that have ${value} ${category}.`
+      )
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+    } else {
+      alert(
+        `No, the person doesn't have ${value} ${category}! Removing all people that have ${value} ${category}.`
+      )
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    }
+  }
 
-  // Invoke a function to redraw the board with the remaining people.
+//Redraws the board with the people left to choose from.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  choosePerson = confirm (`Guess on ${personToConfirm}?`)
+  if (choosePerson) {
+    checkMyGuess(personToConfirm)
+  } 
 }
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name) {
+    winOrLose.style.display = "flex"
+    winOrLoseText.innerText = `
+    That's correct! Good job!
+    `
+    board.innerHTML = ''
+  } else {
+    winOrLose.style.display = "flex"
+    winOrLoseText.innerText = `
+    Wrong answer! Try again!
+    `
+    board.innerHTML = ''
+  }
 }
 
 // Invokes the start function when website is loaded
 start()
 
-// All the event listeners
+// All event listeners
 restartButton.addEventListener('click', start)
+questions.addEventListener('change', selectQuestion)
+filter.addEventListener('click', checkQuestion)
+playAgain.addEventListener('click', start)
```

Comment:
Here you can add a function that removes the winOrLose - like the display = flex you did above to show it, but now display = none and the board display = flex.